model/task_queue_service_dependency.go

```diff
@@ -362,6 +362,61 @@ func (d *basicCachedDAGDispatcherImpl) FindNextTask(ctx context.Context, spec Ta
 				}
 			}
 
+			taskVersion, err := VersionFindOneId(nextTaskFromDB.Version)
```

Comment:
Could we refactor the limit-checking logic for this/generate.tasks limits into a helper function? `FindNextTask` is getting rather long/complicated since it's doing so many last-second checks.